cpp/src/join/hash_join.cu

```diff
@@ -219,23 +287,35 @@ std::size_t get_full_join_size(cudf::table_device_view build_table,
   auto right_indices = std::make_unique<rmm::device_uvector<size_type>>(join_size, stream, mr);
 
   auto const probe_nulls = cudf::nullate::DYNAMIC{has_nulls};
-  pair_equality equality{probe_table, build_table, probe_nulls, compare_nulls};
 
-  row_hash hash_probe{probe_nulls, probe_table};
+  auto const row_hash           = cudf::experimental::row::hash::row_hasher{preprocessed_probe};
+  auto const hash_probe         = row_hash.device_hasher(probe_nulls);
   auto const empty_key_sentinel = hash_table.get_empty_key_sentinel();
-  make_pair_function pair_func{hash_probe, empty_key_sentinel};
-
-  auto iter = cudf::detail::make_counting_transform_iterator(0, pair_func);
+  auto const iter               = cudf::detail::make_counting_transform_iterator(
+    0, make_pair_function{hash_probe, empty_key_sentinel});
 
-  const cudf::size_type probe_table_num_rows = probe_table.num_rows();
+  cudf::size_type const probe_table_num_rows = probe_table.num_rows();
 
-  auto out1_zip_begin = thrust::make_zip_iterator(
+  auto const out1_zip_begin = thrust::make_zip_iterator(
     thrust::make_tuple(thrust::make_discard_iterator(), left_indices->begin()));
-  auto out2_zip_begin = thrust::make_zip_iterator(
+  auto const out2_zip_begin = thrust::make_zip_iterator(
     thrust::make_tuple(thrust::make_discard_iterator(), right_indices->begin()));
 
-  hash_table.pair_retrieve_outer(
-    iter, iter + probe_table_num_rows, out1_zip_begin, out2_zip_begin, equality, stream.value());
+  auto const row_comparator =
+    cudf::experimental::row::equality::two_table_comparator{preprocessed_probe, preprocessed_build};
+  auto const comparator_helper = [&](auto device_comparator) {
+    pair_equality equality{device_comparator};
+
```

Comment:
```suggestion
```

Reply:
I think this line is okay. It's just an empty line that separates the next line, which is a multi-line function call

Reply:
Okay removed it anyway. Pushing the changes